python/lsst/ts/ess/csc/data_client/controller_data_client.py

```diff
@@ -221,44 +217,49 @@ async def connect(self) -> None:
                 # so descr and __repr__ show the correct host and port
                 port = self.mock_server.port
             else:
-                self.log.info(f"{self}.enable_mock_server false; connection will fail")
+                self.log.info(f"{self}.enable_mock_server false; connection will fail.")
                 port = 0
             # Change self.config so descr and __repr__ show the actual
             # host and port.
             self.config.host = tcpip.LOCAL_HOST
             self.config.port = port
 
-        self.reader, self.writer = await asyncio.wait_for(
-            asyncio.open_connection(
-                host=self.config.host, port=self.config.port
-            ),  # type: ignore
-            timeout=CONNECT_TIMEOUT,
+        self.client = tcpip.Client(
+            host=self.config.host,
+            port=self.config.port,
+            log=self.log,
+            name=type(self).__name__,
+        )
+        await asyncio.wait_for(fut=self.client.start_task, timeout=CONNECT_TIMEOUT)
+        configuration = {common.Key.DEVICES: self.config.devices}
+        await self.run_command(
+            command=common.Command.CONFIGURE, configuration=configuration
         )
-        configuration = {"devices": self.config.devices}
-        await self.run_command(command="configure", configuration=configuration)
 
     async def disconnect(self) -> None:
-        """Disconnect from the RPi.
+        """Disconnect from the ESS Controller.
 
         Always safe to call, though it may raise asyncio.CancelledError
-        if the writer is currently being closed.
+        if the client is currently being closed.
         """
         self.run_task.cancel()
         if self.connected:
-            await asyncio.wait_for(
-                tcpip.close_stream_writer(self.writer), timeout=CONNECT_TIMEOUT
-            )
+            assert self.client is not None  # make mypy happy
+            await self.client.close()
```

Comment:
I gently suggest setting self.client to None, as well, to reduce the chances that code can try to use it when it is no longer usable.